fbfmaproom/fbfmaproom.py

```diff
@@ -511,14 +510,16 @@ def select_obs(country_key, obs_keys, target_month0, target_year=None):
         except KeyError:
             raise NotFoundError(f'No value for {" ".join(obs_keys)} on {target_date}') from None
 
+    target_month = int(target_month0) + 1
+    target_day = (target_month0 % 1) * 30 + 1
     with warnings.catch_warnings():
         # ds.where in xarray 2022.3.0 uses deprecated numpy
         # functionality. A recent change deletes the offending line;
         # see if this catch_warnings can be removed once that's
         # released.
         # https://github.com/pydata/xarray/commit/3a320724100ab05531d8d18ca8cb279a8e4f5c7f
         warnings.filterwarnings("ignore", category=DeprecationWarning, module='numpy.core.fromnumeric')
-        ds = ds.where(lambda x: x["time"].dt.month == target_month0 + 0.5, drop=True)
+        ds = ds.where(lambda x: (x["time"].dt.month == target_month) & (x["time"].dt.day == target_day), drop=True)
```

Comment:
I am not sure what ds looks like before the where, but I figure I'd point out that different seasonal configurations can have same center, e.g. JJAS and JA.

Reply:
With the exception of the ENSO dataset, the obs files are all single-season, and this line has no effect. For reasons that in retrospect might not have been very good, we decided to use a single ENSO dataset that's shared by all the countries and seasons, and this line pulls out the relevant season from that special dataset. Then in this PR I introduce a new ENSO-like dataset for four-month seasons instead of three.

Conclusion: seasons of different length are never mixed in the same file, so the risk you pointed out doesn't affect us, but maybe the way we handle ENSO should be revisited.